### PR TITLE
[NFC][SYCL] Refactor Basic/vector/byte.cpp E2E test

### PR DESCRIPTION
**Pull request by uditagarwal97**
to re-arrange code, add asserts, and update variable names to more meaning full ones.

### CODE REVIEW

File: sycl/test-e2e/Basic/vector/byte.cpp [hunk 1]
```diff
@@ -19,7 +19,7 @@
 #include <tuple>   // std::ignore
 
 int main() {
-  std::byte bt{7};
+  std::byte bt{2};
```
Comment (aelovikov-intel) [inline sycl/test-e2e/Basic/vector/byte.cpp:22]: Why this one? I think it is supposed to be different from the value on line 29.

File: sycl/test-e2e/Basic/vector/byte.cpp [hunk 2]
```diff
@@ -92,212 +104,249 @@ int main() {
...
+    sycl::vec<std::byte, 3> VecByte3A{std::byte{4}, std::byte{9},
+                                      std::byte{25}};
+    sycl::vec<std::byte, 3> VecByte3B{std::byte{2}, std::byte{3}, std::byte{5}};
+    sycl::vec<std::byte, 4> VecByte4A{std::byte{5}, std::byte{6}, std::byte{2},
+                                      std::byte{3}};
+
+    // Test bitwise operations on vec<std::byte> and swizzles.
+    // Adding asserts on vec<> operations, and not swizzle operations,
+    // should suffice as swizzles just delegates the operation to vec<>
+    // class.
```
Comment (aelovikov-intel) [inline sycl/test-e2e/Basic/vector/byte.cpp:132]: Is this new or simply stating what was done before? It doesn't sound like a right thing to do to me.

File: sycl/test-e2e/Basic/vector/byte.cpp [hunk 3]
```diff
@@ -92,212 +104,249 @@ int main() {
...
+      assert(VecByte3And[0] == (VecByte3A[0] & VecByte3B[0]));
+      assert(VecByte3Or[1] == (VecByte3A[1] | VecByte3B[1]));
+      assert(VecByte3Xor[2] == (VecByte3A[2] ^ VecByte3B[2]));
+
+      // logical binary op for 2 swizzle
+      auto swand = SwizByte2A & SwizByte2B;
+      auto swor = SwizByte2A | SwizByte2B;
+      auto swxor = SwizByte2A ^ SwizByte2B;
+
+      // Check order of operands for bitwise operators.
```
Comment (aelovikov-intel) [inline sycl/test-e2e/Basic/vector/byte.cpp:150]:
```suggestion
      // Check overloads with a scalar argument for bitwise operators.
```

File: sycl/test-e2e/Basic/vector/byte.cpp (hunk 1 above)
Comment (uditagarwal97) [reply sycl/test-e2e/Basic/vector/byte.cpp:22]: `2` has a simpler bit pattern so, it made it easier to write asserts for `as<>` and `convert<>` method. I can change it to `4`, instead.

File: sycl/test-e2e/Basic/vector/byte.cpp (hunk 2 above)
Comment (uditagarwal97) [reply sycl/test-e2e/Basic/vector/byte.cpp:132]: It's new - I've added new asserts only for `sycl::vec<>` operations, not swizzles.

**Review by aelovikov-intel — APPROVED** [review]

**Pull request merged**